warehouse/utils/project.py

```diff
@@ -61,3 +73,26 @@ def remove_project(project, request, flash=True):
             f"Successfully deleted the project {project.name!r}.",
             queue="success",
         )
+
+
+def destroy_docs(project, request, flash=True):
+    request.db.add(
+        JournalEntry(
+            name=project.name,
+            action="docdestroy",
+            submitted_by=request.user,
+            submitted_from=request.remote_addr,
+        )
+    )
+
+    request.task(remove_documentation).delay(project.name)
+
+    project.has_docs = False
+    # Flush so we can repeat this multiple times if necessary
+    request.db.flush()
```

Comment:
Do we really need a flush here? That really surprises me that it's required.

Reply:
eh, probably not, definitely just copy pasta :)